File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        
        int[] heights = new int[n];
        
        for (int i = 0; i < n; i++) {
            heights[i] = Integer.parseInt(br.readLine());
        }
        Stack<Integer> stack = new Stack<>();
        long answer = 0;
        for(int i =0 ;i<n;i++){
            while(!stack.isEmpty()&& stack.peek()<=heights[i]){
                stack.pop();                
            }
            answer += stack.size();
            stack.push(heights[i]);
        }
        System.out.println(answer);

    }
}

File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.py
n = int(input())
arr = [int(input()) for i in range(n)]

stack=[]
sum =0

for i in range(n):
    while stack and stack[-1]<=arr[i]:
            stack.pop()
    sum+=len(stack)
    stack.append(arr[i])


print(sum)